Gems/Prefab/PrefabDependencyViewer/Code/Tests/PrefabDependencyViewerFixture.h

```diff
@@ -0,0 +1,139 @@
+ *
+ */
+
+#pragma once
+
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzToolsFramework/Prefab/PrefabDomTypes.h>
+#include <MockPrefabSystemComponentInterface.h>
+#include <AzToolsFramework/Prefab/PrefabDomUtils.h>
+
+namespace PrefabDependencyViewer
+{
+    using PrefabDom    = AzToolsFramework::Prefab::PrefabDom;
+    using PrefabDomMap = AZStd::unordered_map<AZStd::string, PrefabDom>;
+
+    struct PrefabDependencyViewerFixture : public UnitTest::ScopedAllocatorSetupFixture
+    {
+        PrefabDependencyViewerFixture()
+            : m_prefabDomsCases(PrefabDomMap())
+            , m_prefabSystemComponent(aznew MockPrefabSystemComponent())
+            {}
+
+        void SetUp() override
+        {
+            UnitTest::ScopedAllocatorSetupFixture::SetUp();
+
+            // Setup for an invalid empty prefab
+            m_prefabDomsCases["emptyJSON"] = PrefabDom();
+            m_prefabDomsCases["emptyJSON"].SetObject();
+
+            // Setup for a root level Prefab with only Source Attribute
+            CreatePrefabAddSourceAndValue("emptyJSONWithSource", "Prefabs/emptySavedJSON.prefab");
+
+            // Setup for root level Prefab with nested instances but one of the nested instances is missing source
+            CreatePrefabAddSourceAndValue("NestedPrefabWithAtleastOneInvalidNestedInstance", "Prefabs/Root.prefab");
+            CreatePrefabAddSourceAndValue("GoodNestedPrefab", "Prefabs/goodPrefab.prefab");
+            CreatePrefabAddSourceAndValue("BadNestedPrefab", "");
+            AddInstance("NestedPrefabWithAtleastOneInvalidNestedInstance", "GoodNestedPrefab");
+            AddInstance("NestedPrefabWithAtleastOneInvalidNestedInstance", "BadNestedPrefab");
+
+            // Setup for valid nested Prefab
+            CreatePrefabAddSourceAndValue("ValidPrefab",   "Prefabs/ValidPrefab.prefab");
+            CreatePrefabAddSourceAndValue("level11Prefab", "Prefabs/level11.prefab");
+            CreatePrefabAddSourceAndValue("level12Prefab", "Prefabs/level12.prefab");
+            CreatePrefabAddSourceAndValue("level13Prefab", "Prefabs/level13.prefab");
+            CreatePrefabAddSourceAndValue("level22Prefab", "Prefabs/level22.prefab");
+            CreatePrefabAddSourceAndValue("level23Prefab", "Prefabs/level23.prefab");
+            CreatePrefabAddSourceAndValue("level31Prefab", "Prefabs/level31.prefab");
+
+            // Level 1 setup
+            AddInstance("ValidPrefab", "level11Prefab");
+            AddInstance("ValidPrefab", "level12Prefab");
+            AddInstance("ValidPrefab", "level13Prefab");
+
+            // Level 2 setup
+            AddInstance("level11Prefab", "level12Prefab");
+            AddInstance("level13Prefab", "level22Prefab");
+            AddInstance("level13Prefab", "level23Prefab");
+
+            // Level 3 setup
+            AddInstance("level23Prefab", "level31Prefab");
+        }
+
+        void CreatePrefabAddSourceAndValue(AZStd::string prefabName, const char* prefabSource)
```

Comment:
Repeated: Since `prefabName` is never copied it's better to pass it by reference `const AZStd::string&`.

Reply:
Resolved